### PR TITLE
FIX: Better error msgs

### LINKED ISSUE #74: AssertionError with missing *-1.fif file
With command 
simulate_movement.py --raw spi_11m_182_raw.fif --pos spi_182.pos --dipoles simulated_dipole_activations.txt --cov simple --out spi_11m_182_simu150717_raw.fif --plot
I get the following error: 

-------------------------------------
Simulate raw data with head movements
-------------------------------------

Reading dipole definitions ....................... done (0.0 sec)
Loading head positions ........................... done (0.0 sec)
Loading raw data file ............................ Split raw file detected but next file /home/staulu/Data/Naja/spi_11m_182_raw-1.fif does not exist.
done (0.1 sec)
Using diagonal covariance for brain noise
Setting up spherical model ....................... done (0.1 sec)
Constructing source estimate ..................... done (0.0 sec)

Traceback (most recent call last):
  File "/home/staulu/anaconda3/bin/simulate_movement.py", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/staulu/mnefun/bin/simulate_movement.py", line 267, in <module>
    run()
  File "/home/staulu/mnefun/bin/simulate_movement.py", line 198, in run
    n_jobs=n_jobs, verbose=True)
  File "<string>", line 2, in simulate_movement
  File "/home/staulu/mne-python/mne/utils.py", line 538, in verbose
    return function(*args, **kwargs)
  File "/home/staulu/mnefun/mnefun/_simulate.py", line 332, in simulate_movement
    assert not (ts > tend).any()
AssertionError

The problem was related to the missing spi_11m_182_raw-1.fif file, but the error message could be clearer.

### LINKED ISSUE #75: RuntimeError: Could not find any MEG or EEG channels
With command 
simulate_movement.py --raw spi_11m_182_raw.fif --pos spi_182.pos --dipoles simulated_dipole_activations.txt --cov simple --out spi_11m_182_simu150717_raw.fif --plot
I get the error of missing MEG channels:

-------------------------------------
Simulate raw data with head movements
-------------------------------------

Reading dipole definitions ....................... done (0.0 sec)
Loading head positions ........................... done (0.0 sec)
Loading raw data file ............................ done (0.2 sec)
Using diagonal covariance for brain noise
Setting up spherical model ....................... done (0.1 sec)
Constructing source estimate ..................... done (0.0 sec)

Provided parameters will provide approximately 2202 events
Setting up raw data simulation using 337 head positions
Setting up forward solutions
Read 1 source spaces a total of 2 active source locations
Traceback (most recent call last):
  File "/home/staulu/anaconda3/bin/simulate_movement.py", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/staulu/mnefun/bin/simulate_movement.py", line 267, in <module>
    run()
  File "/home/staulu/mnefun/bin/simulate_movement.py", line 198, in run
    n_jobs=n_jobs, verbose=True)
  File "<string>", line 2, in simulate_movement
  File "/home/staulu/mne-python/mne/utils.py", line 538, in verbose
    return function(*args, **kwargs)
  File "/home/staulu/mnefun/mnefun/_simulate.py", line 448, in simulate_movement
    chpi_rrs, eog_rr, ecg_rr, n_jobs)):
  File "/home/staulu/mnefun/mnefun/_simulate.py", line 129, in _make_forward_solutions
    _prep_channels(info, False, True, True, verbose=False)
  File "<string>", line 2, in _prep_channels
  File "/home/staulu/mne-python/mne/utils.py", line 538, in verbose
    return function(*args, **kwargs)
  File "/home/staulu/mne-python/mne/forward/_make_forward.py", line 283, in _prep_channels
    raise RuntimeError('Could not find any MEG or EEG channels')
RuntimeError: Could not find any MEG or EEG channels

/neuro/bin/util/show_fiff indicates that there are no obvious problems with the raw data file.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #74.
Closes #75.

@staulu can you try this branch to see if it works for you? You can follow the `git` helper document on Maintainers here:

https://github.com/LABSN/expyfun/blob/master/git_flow.rst#maintainers

If you don't feel like messing with `git`, I can have someone review the changes and merge into `master`, and we can iterate that way.

cc @wronk you will want to make these sorts of changes in your mne-python PR, too. Notably, I didn't do any protection for MEG-only or EEG-only before. This adds checks for MEG-only. In the EEG-only case, we don't even have to simulate head positions, which is nice. Please add a note to yourself somewhere (probably on GitHub in your PR) to make sure MEG-only, EEG-only, and MEG+EEG all work.

### CODE REVIEW

**Comment by wronk** [comment]
Will do, thanks.

**Comment by staulu** [comment]
Thanks Eric, I'll look into this with git tomorrow.

**Comment by staulu** [comment]
With Mark's help I was able to get to the stage where my terminal window claims (after issuing git branch) that I have the branch 'eric/minor-fixes' rather than 'master' in use. I thought that I could then just run the simulation script from the command line with the fixed functions being effective. Obviously, I am doing something wrong as I still get the same error messages as before. Sorry that I am still new to git and don't understand too much about it. Please advice. Thanks!

**Comment by Eric89GXL** [comment]
1. Let's just be extra sure, do `git log` and verify that my last two commits from this PR are at the top of the log. I say this because I would have expected the command you used to be `git checkout -b minor-fixes eric/minor-fixes`, not `git branch` -- but you can get to the proper state using `branch`, too, if you also use some additional commands.

2. How did you install mnefun? Did you do `python setup.py install --user` or `python setup.py develop --user`? If you don't remember, change directory to any directory that *doesn't* have an `mnefun` directory within it, and do `python -c "import mnefun; print(mnefun)". This will list the install path. If you did `install`, it will be something like `/home/samu/.local/lib/python2.7/site_packages/mnefun/__init__.py`, if you used `develop` it will be something like `/home/samu/mnefun/mnefun`. I suspect it will be the latter based on your error report (and thus you must have done `python setup.py develop --user`, but it's worth double-checking.

**Comment by wronk** [comment]
What script do you need to run? Is it internal to mnefun or something only you have on your local machine?

I can come back tomorrow morning to help fix the issue.

**Comment by Eric89GXL** [comment]
@wronk it's a script installed by `mnefun` called `simulate_movement.py`, but this PR only changes `mnefun` module code, not the script itself. In any case I think `develop` usually makes symlinks to the scripts/executables anyway, so even with mods to those we should be okay.

**Comment by Eric89GXL** [comment]
3. Assuming your `git log` is incorrect, can you also do `git remote -v` and give the result?

**Comment by wronk** [comment]
I believe we got the new branch checked out okay after setting up a remote and fetching all of your branches, so I'm suspecting issue 2...

**Comment by Eric89GXL** [comment]
What commands were used, specifically? Do you remember @wronk?

**Comment by wronk** [comment]
IIRC:

`git remote add eric git://github.com/Eric89GXL/mnefun.git`
`git fetch eric`
`git checkout -b minor-fixes eric/minor-fixes`

**Comment by Eric89GXL** [comment]
Yeah that should work, @staulu can you confirm that was the set of commands? You mentioned `git branch` but maybe you meant `git checkout -b`?

**Comment by Eric89GXL** [comment]
Also `git status` should say you have the branch `minor-fixes` not `eric/minor-fixes` in use. I wonder if you did `git branch eric/minor-fixes` which would not work...

**Comment by wronk** [comment]
I was just showing him how to use `git branch` to see the list of branches and which one is currently in use, so I'm guessing that's what he meant. @staulu, I'll be around all day tomorrow, so just email or grab me from upstairs and we can work on this.

**Comment by staulu** [comment]
Sorry, I was a bit unclear. I just meant that after the commands that Mark issued, git branch seemed to indicate that we had switched to the new branch. I took a screenshot of Mark's commands and they were otherwise as Mark listed above, but the last command was git checkout -b eric/minor-fixes. Thanks to both of you for help, I'll talk to Mark tomorrow, and I'm sure we can figure this out.

**Comment by Eric89GXL** [comment]
Unfortunately `git checkout -b eric/minor-fixes` will just create and check out a branch with that name, you need to do `git checkout -b minor-fixes eric/minor-fixes` (note the local branch name `minor-fixes` preceding the remote branch to track) to get it to create, check out, and set to track my branch. So you can do that and `git branch -d eric/minor-fixes` and you should be in the correct state.

FWIW you can even do something else like `git checkout -b emf eric/minor-fixes` to create a local branch named `emf` that contains the same changes as `eric/minor-fixes`. I actually usually use acronyms like that for branches that I make from other people's work so I can keep straight what is mine and what is not.

**Comment by staulu** [comment]
Thanks to both of you. The simulator works now!

**Comment by Eric89GXL** [comment]
Hooray! @wronk can you review this and merge if you're happy?

File: mnefun/_simulate.py [hunk 1]
```diff
@@ -121,15 +122,20 @@ def _make_forward_solutions(info, mri, src, bem, bem_eog, dev_head_ts, mindist,
...
                 meas_id=None, working_dir=os.getcwd(),
                 command_line='', bads=info['bads'])
 
     # Only get the EEG channels here b/c we can do MEG later
-    _, _, eegels, _, eegnames, _ = \
-        _prep_channels(info, False, True, True, verbose=False)
+    if len(pick_types(info, meg=False, eeg=True)) > 0:
+        do_eeg = True
+        _, _, eegels, _, eegnames, _ = \
+            _prep_channels(info, False, True, True, verbose=False)
```
Comment (wronk) [inline mnefun/_simulate.py]: My PR in progress (#2355) will immediately break this once it get's merged since I split up `_prep_channels` :/

Comment (Eric89GXL) [inline mnefun/_simulate.py]: That's fine, I'll fix that separate PR when the time comes

Comment (wronk) [inline mnefun/_simulate.py]: +1 for merge then

**Pull request merged**